### PR TITLE
[OpenVINO-EP] V3.4 Release with OpenVINO 2021.4.2 LTS Release

### PR DESCRIPTION
**Pull request by MaajidKhan**
**Description:** [OpenVINO-EP] V3.4 Release

**Motivation and Context:**

- Making OpenVINO EP compatible with the latest OpenVINO 2021.4.2 Release
- Added windows fix (windows build was broken)
- Added c# Nuget package build fix (c# build was broken)
- Modified Hetero feature logic in the getcapability()

### CODE REVIEW

File: dockerfiles/Dockerfile.openvino-csharp [hunk 1]
```diff
@@ -97,13 +97,14 @@ RUN apt update -y && \
...
     apt-get update -y && \
-    apt-get install -y dotnet-sdk-3.1 && \
+    apt-get install -y dotnet-sdk-5.0 && \
 # Download and build ONNX Runtime
     cd ${MY_ROOT} && \
     git clone --recursive -b ${ONNXRUNTIME_BRANCH} ${ONNXRUNTIME_REPO} && \
     /bin/sh onnxruntime/dockerfiles/scripts/install_common_deps.sh && \
     pip install onnx==1.9 && \
     cd ${MY_ROOT}/onnxruntime && ./build.sh --config Release --update --build --parallel --use_openvino ${DEVICE} --build_nuget --build_shared_lib && \
+    cp ${MY_ROOT}/onnxruntime/build/Linux/Release/Microsoft.ML.OnnxRuntime.Managed* ${MY_ROOT}/onnxruntime/build/Linux/Release/nuget-artifacts && \
```
Comment (oliviajain) [inline dockerfiles/Dockerfile.openvino-csharp:107]: Out of curiosity, why wasn't this required before?

**Review by oliviajain — APPROVED** [review]

File: cmake/CMakeLists.txt [hunk 2]
```diff
@@ -960,7 +960,11 @@ if (WIN32)
         # issued by thrust nonstandard extension used: nameless struct/union
         list(APPEND ORT_WARNING_FLAGS "/wd4201")
         # warning C4800: Implicit conversion from 'X' to bool. Possible information loss
-        list(APPEND ORT_WARNING_FLAGS "/w34800")
+        if (onnxruntime_USE_OPENVINO)
```
Comment (jywu-msft) [inline cmake/CMakeLists.txt:963]: where does this warning get emitted? should this have only been added for openvino targets?
i.e. follow this pattern as example:
https://github.com/microsoft/onnxruntime/blob/908765c860834091c36a5fe73d862e48097fb364/cmake/CMakeLists.txt#L1675
and
https://github.com/microsoft/onnxruntime/blob/908765c860834091c36a5fe73d862e48097fb364/cmake/onnxruntime_providers.cmake#L602

File: dockerfiles/Dockerfile.openvino-csharp (hunk 1 above)
Comment (MaajidKhan) [reply dockerfiles/Dockerfile.openvino-csharp:107]: @oliviajain Earlier there used to be two nuget packages that used to get generated in nuget-artifacts folder, but due to recent changes in nuget build, now only OpenVINO nuget is being generated in nuget-artifacts folder, so we are copying the other one also to the  artifacts folder.

File: cmake/CMakeLists.txt (hunk 2 above)
Comment (MaajidKhan) [reply cmake/CMakeLists.txt:963]: @jywu-msft Hi George. I think, very recently this level 3 warning flag has been added by default /w34800 for WIN32 builds. so, we had to put a check here, to disable it for OpenVINO EP.

we did even try disabling this warning (/wd4800) at a later stage in the cmake and it wasn't effective and not working.

I did even try the way as commented above by you  where I set the warnings to be disabled in cmake and called it out in onnxruntime_providers.cmake for OpenVINO EP and it didn't work either.

Comment (jywu-msft) [reply cmake/CMakeLists.txt:963]: > 

ok. thanks for checking. i'll take a look at it later as well.

**Comment by ricklina90** [comment]
Hi, According this article https://github.com/microsoft/onnxruntime/issues/9271.
I tried to build OpenVINO-EP for C# nuget package again in  Windows environment, but build still failed.
Anyone as an idea of possible solution? Thanks a lot!

error

**Comment by MaajidKhan** [comment]
@oliviajain @jywu-msft I see the release tag 1.10 is removed to this PR?
will it be taken care of while cherry picking the commits for ONNXRT v1.10 release?

**Comment by oliviajain** [comment]
> @oliviajain @jywu-msft I see the release tag 1.10 is removed to this PR? will it be taken care of while cherry picking the commits for ONNXRT v1.10 release?

The PR has already been cherry picked in the release branch. You can see it in the list of commits [here](https://github.com/microsoft/onnxruntime/commit/31a474269e55e8e7282d095ad3084f7ba6b87ed5).

**Pull request merged**